include/rcpputils/filesystem_helper.hpp

```diff
@@ -259,6 +259,15 @@ RCPPUTILS_PUBLIC bool exists(const path & path_to_check);
  */
 RCPPUTILS_PUBLIC path temp_directory_path();
 
+/**
+ * \brief Construct a uniquely named temporary directory, in "parent", with format base_name.XXXXXX
```

Comment:
why wouldn't this use `temp_directory_path` as listed above this function?

Reply:
At least the way we were using it in rosbag2 - on Linux we were creating these temporary directories within the test directory, rather than within the OS's `/tmp` temporary directory (on Windows is was using the global temp directory). I was thinking that the user _could_ call it with `temp_directory_path()` as the `parent`, but it wouldn't be mandatory. The main functionality of the utility is to format a guaranteed unique name for use, to avoid overlaps with existing tmp dirs, no matter where it is used.

Reply:
I've updated the docstrings for both functions to try and make this more clear